lib/spack/spack/build_systems/python.py

```diff
@@ -222,6 +222,15 @@ def install(self, spec, prefix):
         with working_dir(self.build_directory):
             pip(*args)
 
+        # NOTE: Debian, and some other distros, inject `local` into the path in
+        # sysconfig, newer versions of pip apply this to the prefix path as well as the
+        # system path.  This works around it in a way that will let us largely ignore
+        # it.  See: https://github.com/spack/spack/pull/31939
+        local_path = os.path.join(prefix, "local")
+        if os.path.isdir(local_path) and not os.path.isdir(os.path.join(prefix, "lib")):
```

Comment:
I would remove this second check, RHEL puts things in lib64 instead of lib anyway. If `prefix.local` exists that's a good sign that `prefix.lib[64]` doesn't.

Reply:
Fair point, removed.